src/zenml/models/pipeline_run_models.py

```diff
@@ -286,12 +237,13 @@ def generate_filter(
 
             base_filter = operator(base_filter, unlisted_filter)
 
+        from zenml.zen_stores.schemas import (
+            PipelineDeploymentSchema,
+            PipelineRunSchema,
+        )
+
         if self.code_repository_id:
-            from zenml.zen_stores.schemas import (
-                CodeReferenceSchema,
-                PipelineDeploymentSchema,
-                PipelineRunSchema,
-            )
+            from zenml.zen_stores.schemas import CodeReferenceSchema
```

Comment:
Maybe cleaner to move the other import up aswell

Reply:
Since the models are also referenced in the respective schema.py files, the imports are here to avoid a circular dependency and thus can not be moved to the top of the file. Following your suggestion, I still moved them outside of their respective if blocks.